### PR TITLE
generalize smartdevice class and add bulb support for the cli tool

### LINKED ISSUE #48: LB100 Command Line Test Errors
Trying to figure out if I'm missing something or if its a problem with the module. Any time I run the command line test against a light I get the error trace below.  Any thoughts?

./pyhs100 --ip 192.168.1.93
== Couch Lamp - LB100(US) ==
Traceback (most recent call last):
  File "./pyhs100", line 11, in <module>
    load_entry_point('pyHS100==0.2.4.1', 'console_scripts', 'pyhs100')()
  File "/srv/homeassistant/lib/python3.4/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/pyHS100/cli.py", line 34, in cli
    ctx.invoke(state)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/srv/homeassistant/lib/python3.4/site-packages/pyHS100/cli.py", line 62, in state
    click.echo(click.style("Device state: %s" % plug.state,
  File "/srv/homeassistant/lib/python3.4/site-packages/pyHS100/pyHS100.py", line 668, in state
    relay_state = self.sys_info['relay_state']
KeyError: 'relay_state'

### PR DESCRIPTION
**Pull request by rytilahti**
This PR generalizes the SmartDevice class to cater both types of devices through a common API.
The included cli tool is also updated to allow basic functionality, ```--bulb``` will have to be specified currently to control the bulb-specific commands. This may change in the future though.

First step to fix #48.

### CODE REVIEW

File: pyHS100/cli.py [hunk 1]
```diff
@@ -9,16 +9,18 @@
...
 
 
 @click.group(invoke_without_command=True)
 @click.option('--ip', envvar="PYHS100_IP", required=False)
 @click.option('--debug/--normal', default=False)
+@click.option('--bulb', default=False, is_flag=True)
 @click.pass_context
-def cli(ctx, ip, debug):
+def cli(ctx, ip, debug, bulb):
     """A cli tool for controlling TP-Link smart home plugs."""
```
Comment (kirichkov) [inline pyHS100/cli.py:24]: ... smart home plugs, switches and bulbs.

**Review by kirichkov — APPROVED** [review]
👍

File: pyHS100/cli.py (hunk 1 above)
Comment (rytilahti) [reply pyHS100/cli.py:24]: Yeah, I noticed that I also pushed currently unused DeviceType enum in. This current way is not very nice, but we'll see after we get some test reports into what direction to go from here.

Comment (rytilahti) [reply pyHS100/cli.py:24]: Ungh, sorry, forgot to fix this. Will try to remember to do it next time there will be code cleanups.

**Pull request merged**